tests/python_package_test/test_engine.py

```diff
@@ -1316,10 +1316,10 @@ def has_interaction(treef):
 
         return not has_interaction_flag.any()
 
-    for test_with_categorical_variable in [True, False]:
-        trainset = generate_trainset_for_monotone_constraints_tests(
-            test_with_categorical_variable
-        )
+    trainset = generate_trainset_for_monotone_constraints_tests(
+        test_with_categorical_variable
+    )
+    for test_with_interaction_constraints in [True, False]:
```

Comment:
I'm ok with this change, but could you update the assert on line 1334 with an error message that will tell us which case it failed on if it fails?

```python
error_msg = f"Model not correctly constrained (test_with_interaction_constraints={test_with_interaction_constraints})"

assert is_correctly_constrained(
    constrained_model, test_with_categorical_variable
), error_msg
```

Without that, that particular test could fail and we wouldn't know which case it failed on just from the logs.

I think it would also be useful too include that value of `monotone_constraints_method` in that message too (since it is similarly from an inner `for` loop), but that's not directly related to this PR so up to you if you want to do that at the same time or separately.

Reply:
Good point! Done in 44454821f1c5624fe5ed809f048e47adf47112e1.